internal/k3d/terraform.go

```diff
@@ -53,3 +58,50 @@ type GithubTerraformEnvs struct {
 	AwsAccessKeyId        string
 	AwsSecretAccessKey    string
 }
+
+func TerraformPrep(config *K3dConfig) error {
+
+	path := config.GitopsDir + "/terraform"
+	log.Info().Msgf("Repooo is %s", path)
+	err := filepath.Walk(path, detokenizeterraform(path, config))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func detokenizeterraform(path string, config *K3dConfig) filepath.WalkFunc {
+	return filepath.WalkFunc(func(path string, fi os.FileInfo, err error) error {
+
+		if fi.IsDir() {
+			return nil
+		}
+
+		matched, _ := filepath.Match("*", fi.Name())
+
+		if matched {
+
+			read, err := os.ReadFile(path)
+			if err != nil {
+				return err
+			}
+
+			replacer := strings.NewReplacer(
+				"<ADMIN_TEAM>", config.AdminTeamName,
+				"<DEVELOPER_TEAM>", config.DeveloperTeamName,
+				"<METAPHOR_REPO_NAME>", config.MetaphorRepoName,
+				"<GIT_REPO_NAME>", config.GitopsRepoName,
+			)
+
+			newContents := replacer.Replace(string(read))
+
+			err = os.WriteFile(path, []byte(newContents), 0)
+			if err != nil {
+				return err
+			}
+
+		}
+		return nil
+	})
+
+}
```

Comment:
Same as above, this seems like a verbatim copy of the other functions created elsewhere, why not just call them from a package instead?

Reply:
Issue is in the one used for other clouds,the config parameter is different ,so i will then need to pass seven variables i want to remove the duplication

Reply:
It seems the parameter "type" is different, but you can probably tell they're both just a set of strings:

```go
func detokenizeterraform(path string, clctrl *ClusterController) filepath.WalkFunc {

// ...

		replacer := strings.NewReplacer(
			"<ADMIN_TEAM>", clctrl.AdminTeamName,
			"<DEVELOPER_TEAM>", clctrl.DeveloperTeamName,
			"<METAPHOR_REPO_NAME>", clctrl.MetaphorRepoName,
			"<GIT_REPO_NAME>", clctrl.GitopsRepoName,
		)
```

and...

```go
func detokenizeterraform(path string, config *K3dConfig) filepath.WalkFunc {

// ...

			replacer := strings.NewReplacer(
				"<ADMIN_TEAM>", config.AdminTeamName,
				"<DEVELOPER_TEAM>", config.DeveloperTeamName,
				"<METAPHOR_REPO_NAME>", config.MetaphorRepoName,
				"<GIT_REPO_NAME>", config.GitopsRepoName,
			)
```

So the generic function could just receive a list of mappings, for example, or the parameters individualized.

My callout here is that if we ever need to introduce a change, not only we have to remember which parts have this replacer, but also since we don't have unit tests for these, another developer changing stuff might inadvertently break one of the two (or three) functions. 

Happy to help you make it happen!